src/Microsoft.Extensions.Hosting/Internal/Host.cs

```diff
@@ -8,22 +8,31 @@
 using System.Threading.Tasks;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Extensions.Hosting.Internal
 {
     internal class Host : IHost
     {
-        private ILogger<Host> _logger;
-        private IHostLifetime _hostLifetime;
-        private ApplicationLifetime _applicationLifetime;
+        private readonly ILogger<Host> _logger;
+        private readonly IHostLifetime _hostLifetime;
+        private readonly ApplicationLifetime _applicationLifetime;
+        private readonly HostOptions _options;
         private IEnumerable<IHostedService> _hostedServices;
 
-        internal Host(IServiceProvider services)
+        public Host(IServiceProvider services, IApplicationLifetime applicationLifetime, ILogger<Host> logger,
+            IHostLifetime hostLifetime, IOptions<HostOptions> options)
         {
+            if (options == null)
```

Comment:
Nit: Might be worth guarding agains and NRE in `StopAsync` by ensuring `options.Value` isn't null. ConsoleLifetime does `options?.Value ?? throw new ArgumentNullException(nameof(options));` which I think is reasonable enough.